core/trino-main/src/main/java/io/trino/execution/scheduler/StageTaskSourceFactory.java

```diff
@@ -480,9 +484,11 @@ public void close()
         private final int splitBatchSize;
         private final LongConsumer getSplitTimeRecorder;
         private final Optional<CatalogName> catalogRequirement;
-        private final int targetPartitionSplitCount;
+        private final int minPartitionSplitCount;
+        private final long targetPartitionSplitWeight;
+        private final int maxPartitionSplitCount;
 
-        private final Queue<Split> remotelyAccessibleSplitBuffer = new ArrayDeque<>();
+        private final Set<Split> remotelyAccessibleSplitBuffer = newIdentityHashSet();
```

Comment:
@martint Provided a reasonable suggestion on the other thread to declare variables as `Identity...` if it is important. Could you please declare the `remotelyAccessibleSplitBuffer` as such and maybe change the `locallyAccessibleSplitBuffer` declaration as well.

Reply:
There is no `IdentityHashSet` class/interface :)

Reply:
Oh, right. There's only IdentityHashMap. Nevermind.